Utilities/Files.cs

```diff
@@ -237,4 +277,53 @@ public static bool IsFileLocked(FileInfo file)
             return false;
         }
     }
+
+    // Code from https://www.codeproject.com/Tips/1190802/File-Locking-in-a-Multi-Threaded-Environment
+    // The code maintains a concurrent dictionary of lock objects and keeps track of how many calls are in the queue for
+    // accessing each filename. If the count goes down to 0, the key is removed from the dictionary as a cleanup mechanism.
+    class LockManager
+    {
+        private static readonly ConcurrentDictionary<string, lobj> _locks =
+                new ConcurrentDictionary<string, lobj>();
+
+        public static void GetLock(string filename, Action action)
+        {
+            lock (GetLock(filename))
+            {
+                action();
+                Unlock(filename);
```

Comment:
Just curious: are you sure that `Unlock(filename)` will get executed if action throws?

Reply:
Good point. I'll wrap it with try-finally to ensure that `Unlock` will execute regardless of any throw.